File: 1000/479A/main.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    int a,b,c;
    cin>>a;
    cin>>b;
    cin>>c;
    
    int maxi = a+b+c;
    maxi = max(maxi, a+b*c);
    maxi = max(maxi, a*b*c);
    maxi = max(maxi, (a+b)*c);
    maxi = max(maxi, a*b+c);
    maxi = max(maxi, a*(b+c));
    
    cout<<maxi;
    
    return 0;
}

File: 1000/479A/main.java
import java.util.*;

public class main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);

        int a = sc.nextInt();
        int b = sc.nextInt();
        int c = sc.nextInt();

        int maxi = a + b + c;
        maxi = Math.max(maxi, a + b * c);
        maxi = Math.max(maxi, a * b * c);
        maxi = Math.max(maxi, (a + b) * c);
        maxi = Math.max(maxi, a * b + c);
        maxi = Math.max(maxi, a * (b + c));

        System.out.println(maxi);
    }
}
